pacai/agents/ghost/directional.py

```diff
@@ -47,7 +48,10 @@ def getDistribution(self, state):
             dist[a] = float(bestProb) / len(bestActions)
 
         for a in legalActions:
-            dist[a] += float(1 - bestProb) / len(legalActions)
-
-        dist.normalize()
+            if not a in dist:
+                dist[a] = float(1 - bestProb) / len(legalActions)
+            else:
+                dist[a] += float(1 - bestProb) / len(legalActions)
```

Comment:
I don't mean an if/else here.
Take a closer look at my suggested structure in the last comment.
Part of the reason for it is that the actual computation code will only happen once, so it is easier to manage in the future
(e.g. if there was a bug with the computation we wouldn't have to worry about accidentally only fixing it on one line).

Reply:
Ok, so my below change perdoem the computation only once.
I don't beleive not doing if/else and only using if is a great idea, since doing it that way may end up performing the set and then and addition operations in the same loop